### PR TITLE
Extend help with possible variables

### LINKED ISSUE #68: List all possible outputs in the CLI and docs
To easier understanding of report and better scanner integration it would be useful to list all possible states in which reported fields can be

For example:
```
rule: str, any value

severity: str
- "critical"
- "high"
- "medium"
- "low"
- "info"

ml_validation: str
- "INVALID_KEY"
- "VALIDATED_KEY"
- "UNDECIDED"
- "NOT_AVAILABLE"

entropy_validation: bool
- true
- false

...
```

Why this is necessary?
For example for now we have `"critical"` severity in code, but no actual rule with this severity
Is we would add a new rule with `"critical"` severity it might brake systems that don't expect severity to be anything else but `high/medium/low/info` 

Where to list?
Good place to mention possible outputs might be CLI `--help` and [how to use doc section](https://credsweeper.readthedocs.io/en/latest/guide.html)

### PR DESCRIPTION
**Pull request by babenek**
## Description

Please include a summary of the change and which is is fixed.

- Added extended help
- fixes #68 

## How has this been tested?

Please describe the tests that you ran to verify your changes.

- [x] invoke credsweeper with --help

### CODE REVIEW

**Review by csh519 — APPROVED** [review]
Adding more information to `help`.

LGTM :+1:

**Review by iuriimet — APPROVED** [review]
Looks good

File: docs/source/guide.rst [hunk 1]
```diff
@@ -27,7 +27,7 @@ Get all argument list:
                             exporting default config to file (default: config.json)
       --export_log_config [PATH]
                             exporting default logger config to file (default: log.yaml)
-      --rules [PATH]        path of rule config file (default: credsweeper/rules/config.yaml)
+      --rules [PATH]        path of rule config file (default: credsweeper/rules/config.yaml). severity:['critical', 'high', 'medium', 'low', 'info'] type:['keyword', 'pattern', 'pem_key']
```
Comment (kmnls) [inline docs/source/guide.rst:30]: Why are these descriptions merged together?
--rules are about the file path, not file content details.

Comment (babenek) [reply docs/source/guide.rst:30]: There was a question about severity... Possibly it might be removed in future

**Pull request merged**